File: 22- Jump to last index/Java/solution3.java
// By keeping track of the max index that we can reach:

// Time complexity: O(n)
// Space complexity: O(1)


static boolean canJump(int[] arr){
  int n = arr.length;
  int maxIndex = 0;
  for(int i = 0; i < n; i++){
    if(i > maxIndex) return false;
    else maxIndex = Math.max(maxIndex, i+arr[i]);
  }
  return maxIndex >= n-1;
}

File: 6- Maximum subarray/C++/solution3.cpp
// By using Kadane's algorithm:

// Time complexity: O(n)
// Space complexity: O(1)

int maximumSubarray(vector<int> arr){
  int globalSum = numeric_limits<int>::min();
  int localSum = 0;
  for(int i = 0; i < arr.size(); i++){
    localSum = max(arr[i], localSum + arr[i]);
    globalSum = max(globalSum, localSum);
  }
  return globalSum;
}


